arch/xtensa/src/esp32/esp32_allocateheap.c

```diff
@@ -96,8 +96,15 @@ void xtensa_add_region(void)
 {
   size_t region2_start = HEAP_REGION2_START + XTENSA_IMEM_REGION_SIZE;
 
+#ifndef CONFIG_SMP
   umm_addregion((FAR void *)region2_start,
                 (size_t)(uintptr_t)&_eheap - region2_start);
+#else
+  umm_addregion((FAR void *)region2_start,
+                (size_t)HEAP_REGION2_END - region2_start);
+  umm_addregion((FAR void *)HEAP_REGION3_START,
+                (size_t)&_eheap - HEAP_REGION3_START);
```

Comment:
this region3 stuff seems to belong to the other commit ("ESP32: Re-organise the different heap regions"), not this one ("arch/xtensa/esp32: Part of the ROM regions in middle of DRAM are not")

Reply:
you can history-edit with git to make it cleaner if you want. but i'm not sure if it's worth the effort. it's up to you.
my suggestion is do nothing.

Reply:
I was going to edit the commit, but I remembered that it was part of it in the first place.

> my suggestion is do nothing.

I'll take your suggestion. ;)